Objects/unicodeobject.c

```diff
@@ -4710,7 +4712,25 @@ ascii_decode(const char *start, const char *end, Py_UCS1 *dest)
         /* Help allocation */
         const char *_p = p;
         Py_UCS1 * q = dest;
-        while (_p + SIZEOF_SIZE_T <= end) {
+        while (_p <= unroll_end) {
+            const size_t *restrict __p = (const size_t *)_p;
```

Comment:
Is the restrict keyword and the assignment to value0, ..., value3 required for the compiler? If not, then this code can be written more compact.

Reply:
In general I think the compiler can judge from the function that no writes happen in either `__p` or `_q` so pointer aliasing is not problematic. But I'd rather make this explicit rather than relying on the compiler to deduce this. MSVC and Clang do this scalarly despite the hint. GCC correctly vectorizes the load store and does not need the hint. I cannot judge all possible compilers out there. I think it is more likely that clang and MSVC will do the correct thing in the future with the hint than without it. 

It would only save the `restrict` keyword as the `(size_t *)` cast is still needed, as that makes the rest of the code more readable. Are the 8 characters extra width that problematic?